### PR TITLE
Change all Proof-of-Work references to Proof-of-Woof

### PR DESCRIPTION
**Pull request by willwooten**
In order to signal our support for Elon and join the fight against wasteful energy consumption, this PR changes all references to dirty proof--of-work to "proof-of-wow"

### CODE REVIEW

**Comment by Cyclic** [comment]
> In order to signal our support for Elon and join the fight against wasteful energy consumption, this PR changes all references to dirty proof--of-work to "proof-of-wow"

Allow me to elaborate. This is what happens when a bunch of jealous trolls have access to something that inspires, changes, and innovates for good. Similar to when GOdDARD created the original vision for rockets, and the majority of Americans were still anti-science clowns laughed at him, until he won the war against the Nazis and shipped the first multistage, liquid fueled rocket.

Egod, I mean Doge, is the first energy efficient and fast implantation of a decentralized currency. Until a solid proof of stake, non-centralized system, is in production, fully vetted, and fully tested — we are sticking with rocket-bound pups.

If you have nothing to add, but clown behavior, then maybe fork this and ship your own rockets. But if you’re just adding more wasteful entropy to the universe, please do it somewhere with less value before presenting it as something good. Good is the key.

thank you,
The light 💡

**Comment by Shandelier** [comment]
Very science wow. Much approval

**Review by PapaOscar90 — APPROVED** [review]
Only wows. Legit wow.

**Comment by StrobFX** [comment]
I would sugget "Proof-of-Woof" instead!

**Comment by PapaOscar90** [comment]
> 
> 
> I would sugget "Proof-of-Woof" instead!

Much approve.

**Comment by StrobFX** [comment]
Also don't do just replace "work" by "wow" but:
proof-of-work => proof-of-woof
proof of work => proof of woof
proofofwork => proofofwoof

**Review by cchiramberro — APPROVED** [review]

**Review by cchiramberro — APPROVED** [review]

File: src/primitives/block.h [hunk 1]
```diff
@@ -66,7 +66,7 @@ class CBlockHeader : public CPureBlockHeader
 class CBlock : public CBlockHeader
 {
 public:
-    // network and disk
```
Comment (andrepcg) [inline src/primitives/block.h:69]: I can see you did a "replace all" but maybe we should not replace the word "network" with "netwow"

**Comment by EverythingWillBeCode** [comment]
i prefer Proof-of-Power

File: src/primitives/block.h (hunk 1 above)
Comment (willwooten) [reply src/primitives/block.h]: done

**Comment by willwooten** [comment]
i am going to get this to run successfully even if I have to sell my hodl to pay the bills in the meantime

**Comment by willwooten** [comment]
success.

now running dogecoind with proof-of-wow

**Comment by gustmp** [comment]
PoWoof

**Review by nopara73 — CHANGES_REQUESTED** [review]
NACK. This makes it harder to port changes upstream from Bitcoin's code base.

**Comment by gabaod** [comment]
why would anyone approve these changes?  you are all over the place when naming variables, some uppercase some lowercase... set a standard and stick to it.. Id delete this PR and start over.

**Comment by willwooten** [comment]
> NACK. This makes it harder to port changes upstream from Bitcoin's code base.

well we can't use bitcoin's code base anymore because Elon is depending on doge core devs to remove wasteful energy use and PoW. did y'all not get the check in the mail? https://twitter.com/elonmusk/status/1392974251011895300

> why would anyone approve these changes? you are all over the place when naming variables, some uppercase some lowercase... set a standard and stick to it.. Id delete this PR and start over.

upper/lower is the same as before. maybe 1 or 2 accidental changes.

**Comment by delbonis** [comment]
@willwooten 

>well we can't use bitcoin's code base anymore

You'd have to throw away pretty much all of Dogecoin then because Dogecoin is 90+% Bitcoin's and Litecoin's code.  And what do you propose as an alternative that doesn't involve massively changing the architecture of the project?

**Comment by gabaod** [comment]
renaming variables doesnt change the functionality of them.. this is still POW youre just trying to be clever and name it something else, like calling an apple and orange instead.

**Comment by willwooten** [comment]
Updated upper/lowercases as requested

> You'd have to throw away pretty much all of Dogecoin then because Dogecoin is 90+% Bitcoin's and Litecoin's code. And what do you propose as an alternative that doesn't involve massively changing the architecture of the project?

i was unaware of this thank you.

Proof-of-Wow isn't Proof-of-Work. That is the proposal.

**Comment by willwooten** [comment]
Changed all references to Proof-of-Woof

**Comment by Tracachang** [comment]
This change it is one of the best actions I've ever seen to fight climate change, please merge this PR as soon as possible!

**Comment by AlejandroReBa** [comment]
Where are the unit tests that prove you are saving the world?

**Comment by willwooten** [comment]
> Where are the unit tests that prove you are saving the world?

we'd need an oracle to input global temperature data

**Review by Tracachang — APPROVED** [review]

**Review by DaWe35 — APPROVED** [review]
LGTM

**Comment by gabaod** [comment]
If you guys care about Doge, you sure are making it look extremely bad.  If you dont know how to develop, or how to create proper pull requests, all you are doing is hurting the project ;)   Once again, renaming variables DOES NOT change the way this functions.. its STILL PROOF OF WORK.   so unless you want to show Elon what type of folks are trying to update this, you are much better off deleting this PR and actually implement this into PoS or a brand new type of blockchain that doesnt require energy consumption like PoW.  If the Dev behind this github account merged this into MAIN..  id consider this project beyond dead than as this is not one step forward to help make Doge more energy efficient, but steps backwards.

File: doc/release-notes/bitcoin/release-notes-0.9.0.md [hunk 2]
```diff
@@ -175,15 +175,15 @@ RPC:
...
+- Add 'chainwoof' (the total woof done by all blocks since the genesis block)
   to 'getblock' output
 - Make RPC password resistant to timing attacks
 - Clarify help messages and add examples
 - Add 'getrawchangeaddress' call for raw transaction change destinations
 - Reject insanely high fees by default in 'sendrawtransaction'
 - Add RPC call 'decodescript' to decode a hex-encoded transaction script
 - Make 'validateaddress' provide redeemScript
-- Add 'getnetworkhashps' to get the calculated network hashrate
+- Add 'getnetwoofhashps' to get the calculated network hashrate
```
Comment (DavidBurkett) [inline doc/release-notes/bitcoin/release-notes-0.9.0.md:186]: Our Dear Leader hasn't yet deemed networks evil. Let's leave this line alone unless we receive further instructions from Him.

**Comment by willwooten** [comment]
> If you guys care about Doge, you sure are making it look extremely bad

Proof-of-Woof is funny tho

File: doc/release-notes/bitcoin/release-notes-0.9.0.md (hunk 2 above)
Comment (willwooten) [reply doc/release-notes/bitcoin/release-notes-0.9.0.md]: fixed.

Comment (DavidBurkett) [reply doc/release-notes/bitcoin/release-notes-0.9.0.md]: Much fast. Such wow.

**Comment by girldrinksgasoline** [comment]
> @willwooten
> 
> > well we can't use bitcoin's code base anymore
> 
> You'd have to throw away pretty much all of Dogecoin then because Dogecoin is 90+% Bitcoin's and Litecoin's code. And what do you propose as an alternative that doesn't involve massively changing the architecture of the project?

take a bunch from qtum, a utxo chain (a lot of Bitcoin code) with added PoS, an account attraction layer and an evm compatible vm. no clue how difficult or even if possible but much wow if it were

**Comment by coblee** [comment]
I only read the title, but LGTM!

**Comment by gabaod** [comment]
I understand your dedication and that youve invested many hours trying to help out, but the thing is just to rename some variable, and create a whole new concept with absolutely no backing to how it will be any different proof-of-concept, doesnt help this project at all.  You have 18 different commits.. when honestly this should be 1 commit - and it should have taken you less than 5 minutes to implement with a simple sed -i 's/change/changeto/' /dogecoin/*  thus it shows you are way over your head and understand this is constructive criticism and if you really cared about this project you would delete this pull request and take a step away from github but still give support.

**Comment by willwooten** [comment]
this is hilarious

**Comment by willwooten** [comment]
sigh...i have failed to pass the checks.

good times were had though. 

anyone that takes doge seriously, lol.

**Review by slush0 — APPROVED** [review]

**Comment by keblek** [comment]
ACK, with this new variable name Doge will finally become the store of value, medium of exchange AND unit of account that the world both needs and deserves. Thank you for your contribution @willwooten, much applause. I look forward to your future variables renamings.

**Comment by RocksteadyTC** [comment]
This is the dumbest shit Ive ever seen.

**Review by okasi — COMMENTED** [review]
LGTM 🚀

**Comment by haxpor** [comment]
Not to disrespect, I do understand doge attracts mass users, but other than enthusiastic about holding doge for investment, this place should be given non-joke but only full serious development-oriented attention. Things running behind to serve those DeFi, and this wallet have much greater things to be concerned about regarding to performance, security, etc to name a few. If majority of developers enjoy joking and sending funny PRs (even enjoy approving it) then the moment of rugpulls, or security hole we all saw in those DeFi space could creep in to this project.

I believe it's better for the project's outlook and to continue attracting new well-intended contributors by keeping your enthusiastic elsewhere **outside** of this project. With all due respect.

**Comment by willwooten** [comment]
it's extremely sad that devs in here take doge seriously and think my search and replace all was a serious attempt at devolopment. jfc.

the code should be absurd as possible, regardless if it makes it hard to copy bitcoin's updates. unless the maintainers actually do something doge will be taken by musk et al and made a corporate toy.

that news story came out that "the devs" aka the 4 maintainers of this github were in contact with Elon since 2019. Great chance to fill your bags knowing a billionaire would start pumping them.

The spirit of doge is absurd. fuck investment.

**Comment by willwooten** [comment]
the way things are moving, doge will be a chuck-e-cheese token for Mark Cuban and Elon.

**Comment by delbonis** [comment]
>it's extremely sad that devs in here take doge seriously and think my search and replace all was a serious attempt at devolopment. jfc.

https://en.wikipedia.org/wiki/Poe%27s_law

There's been too much going on like this it's impossible to discern satire anymore.

**Comment by willwooten** [comment]
> There's been too much going on like this it's impossible to discern satire anymore.

isn't doge satire itself?

**Comment by davidmroth** [comment]
> > There's been too much going on like this it's impossible to discern satire anymore.
> 
> isn't doge satire itself?

You're probably right, because I was laughing all the way to the bank! 💲💲💲

**Comment by willwooten** [comment]
> You're probably right, because I was laughing all the way to the bank! heavy_dollar_signheavy_dollar_signheavy_dollar_sign

congrats dumping a copy pasta of bitcoin on teenagers

**Comment by trasherdk** [comment]
This is just beautiful. Entertainment of highest caliber, just like this coin.
Thanks for the laughs.

**Comment by bernTOAST** [comment]
I have invented a consensus tx approval mechanism inspired by your proof of work and xrp validator system.  Doge 2.0 rough copy lol

**Pull request closed without merging**